### PR TITLE
Change way of sensebox/home decoder

### PR DESCRIPTION
**Pull request by mpfeil**
`sensebox/home` profile is now able to handle every sensor setup of a **senseBox:home**.
It`s not restricted to the old full sensor setup. If the `payload is longer than the `sensorMatching` of the correspondig box the unused bytes are ignored. 

## ToDo
- [ ] add test where the `payload` is shorter than the `sensorMatching`

### CODE REVIEW

File: lib/decoding/index.js [hunk 1]
```diff
@@ -50,19 +51,22 @@ const bufferToMeasurements = function bufferToMeasurements (buffer, bufferTransf
...
 
   // feed each bufferTransformer element
   for (const mask of bufferTransformer) {
     const maskedBytes = buffer.slice(currByte, currByte + mask.bytes);
 
-    result.push({
-      sensor_id: mask.sensorId,
-      value: mask.transformer(maskedBytes)
-    });
+    if (maskedBytes.length !== 0) {
```
Comment (noerw) [inline lib/decoding/index.js]: Shouldn't this check `maskedBytes.length === mask.bytes` ?

File: lib/decoding/index.js [hunk 2]
```diff
@@ -72,7 +76,7 @@ const bufferToMeasurements = function bufferToMeasurements (buffer, bufferTransf
     }
   }
 
-  return result;
+  return [result, warnings];
```
Comment (noerw) [inline lib/decoding/index.js]: why not `{ result, warnings }`? It's more explicit and same amount of chars ;)

File: lib/decoding/index.js [hunk 3]
```diff
@@ -169,9 +176,18 @@ const decodeRequest = function decodeRequest (req) {
...
+        .then(data => {
+          return [data];
+        });
+    }
+
+    return Promise.reject('no payload for profile `json` provided');
+
+    // return req.body.payload_fields
+    //   ? decodeJSON(req.body.payload_fields)
+    //   : Promise.reject('no payload for profile `json` provided');
```
Comment (noerw) [inline lib/decoding/index.js]: These comments can be deleted

**Review by noerw — COMMENTED** [review]
Few questions:
- This assumes that measurements are submitted in a specific order, correct? Which order? Would be good to document that.
- Shouldn't the case of to few bytes still be treated as an error? When measurements are missing this is not the behaviour as configured.. Also we don't know which ones are missing, and this can mess up the data.
- How are these warnings handled? afaik there is no mechanism to show them to the user..
- Maybe rebase this onto `feature/mqtt`, as that's the latest code.. But can be done later as well I guess

**Comment by mpfeil** [comment]
> This assumes that measurements are submitted in a specific order, correct? Which order? Would be good to document that.

Yes of course. If we merge the PR there will also be a generated [LoRa sketch](https://github.com/sensebox/node-sketch-templater/blob/lora/templates/homev2_lora.tpl) at the end of registration. But afaik the specific order is always relevant. Or did I miss something?

> Shouldn't the case of to few bytes still be treated as an error? When measurements are missing this is not the behaviour as configured.. Also we don't know which ones are missing, and this can mess up the data.

Fair point. I will think about it.

> How are these warnings handled? afaik there is no mechanism to show them to the user..

Right now there is no mechanism. But there will be later. Just as a preparation for it.

> Maybe rebase this onto feature/mqtt, as that's the latest code.. But can be done later as well I guess

We will do it later. We want to release the new LoRa sketch generation and after it release your changes of feature/mqtt

File: lib/decoding/index.js (hunk 1 above)
Comment (mpfeil) [reply lib/decoding/index.js]: Good catch. Resolved with last commit

File: lib/decoding/index.js (hunk 2 above)
Comment (mpfeil) [reply lib/decoding/index.js]: Also changed with the last commit

**Comment by mpfeil** [comment]
Closed. Already included in new `main` branch

**Pull request closed without merging**